### DISCUSSION TITLE
Frequency Manager: UTC start/end for bookmarks

### DISCUSSION THREAD

**Discussion by darauble** [AlexandreRouma/SDRPlusPlus#777] [Ideas]
Hi all,

I wonder if adding UTC start/end times to bookmark would be a big effort? And adding a checkbox to filter out currently "inactive" bookmarks? UTC start/end should be optional, of course, but if they are present, this filtering could be useful. Especially for shortwave listeners, where stations share the ether or broadcast actually goes on/off.

- Suggested format for time entries HHMM, e.g. 0800.
- Basic check for hour/minute range (0-23, 0-59) is also important.
- Start/end: either both are empty or both filled.
- Some intelligence required for midnight crossing times, e.g. start 2300, end 0100.

As a software engineer myself, I could contribute. In theory, as I'm no good with UI or C++, but this feature does not seem very complicated. Depends if author and other users would be interested in this feature.

**Comment by OttoPattemore** [comment]
Hi @darauble,

I agree this would be a very useful feature.  It shouldn't be that hard to implement.
I have done something very similar in my SDR++ plugin [Shortwave Station List](https://github.com/OttoPattemore/shortwave-station-list-sdrpp).  The only difference is that my plugin fetches the bookmarks from a remote server.  The code of interest to you would be [main.cpp line 35](https://github.com/OttoPattemore/shortwave-station-list-sdrpp/blob/main/src/main.cpp#L35)

**Reply by darauble** [reply]
Thank you! That's a great reply and, actually, a great idea to make a plugin instead of SDR++ modification. I think I might then implement this and [my other wish](https://github.com/AlexandreRouma/SDRPlusPlus/issues/533) in a separate plugin.
